### PR TITLE
Support to write js expression that refer to column groups

### PR DESCRIPTION
**Pull request by tangrammer**
This PR add support to write js expressions that could refer to column groups
eg: 
`row['Metadata']['Identifier']` 
`row['Transformations']['my example generated']` 
`row['Flow Question Group1']['my question']` 

Follows up #2450 
Relates https://github.com/akvo/akvo-lumen/issues/2437#issuecomment-559015407

- [ ] **Update release notes if necessary**

### CODE REVIEW

File: backend/src/akvo/lumen/lib/transformation/derive.clj [hunk 1]
```diff
@@ -43,27 +43,32 @@
   or row[\"foo\"]. For every reference return a tuple with matched pattern and
   the row column as in [\"row.foo\" \"foo\"]."
   [code]
-  (let [re #"(?U)row.([\w\d]+)|row(?:\[')([^']*)(?:'\])|row(?:\[\")([^\"]*)(?:\"\])"
+  (let [re #"(?U)row.([\w\d]+)|row(?:\[')([^']*)(?:'\])(?:\[')([^']*)(?:'\])|row(?:\[')([^']*)(?:'\])|row(?:\[\")([^\"]*)(?:\"\])(?:\[\")([^\"]*)(?:\"\])|row(?:\[\")([^\"]*)(?:\"\])"
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation/derive.clj]: I tried but didn't find a better way to do this regex

File: backend/src/akvo/lumen/lib/transformation/derive.clj [hunk 2]
```diff
@@ -43,27 +43,32 @@
...
 
 (defn compute-transformation-code
   "analyses code to find columns relations between column-title and column-name when using js code"
-  [code columns]
-  (let [columns (walk/keywordize-keys columns)]
-    (reduce (fn [m [pattern column-title]]
+  [code columns & [kw?]]
+  (let [columns (if kw? columns
+                    (walk/keywordize-keys columns))]
+    (reduce (fn [m [pattern column-ref1 column-ref2]]
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation/derive.clj]: now, the regex can return 2 column refs, one for the group, one for the column

File: backend/src/akvo/lumen/lib/transformation/derive.clj (hunk 1 above)
Comment (kardan) [reply backend/src/akvo/lumen/lib/transformation/derive.clj]: Oh my :-)

File: backend/src/akvo/lumen/lib/transformation/engine.clj [hunk 3]
```diff
@@ -108,10 +108,13 @@
...
   (format "d%s" i))
 
+(defn is-derivated? [column-name]
+  (boolean (re-find #"^d\d+$" column-name)))
+
 (defn next-column-index [columns]
   (let [nums (->> columns
                   (map #(get % "columnName"))
-                  (filter #(re-find #"^d\d+$" %))
+                  (filter is-derivated?)
```
Comment (kardan) [inline backend/src/akvo/lumen/lib/transformation/engine.clj]: "is-derived?" ?

**Pull request merged**